rust/operator-binary/src/controller.rs

```diff
@@ -467,6 +468,47 @@ fn build_node_rolegroup_service(
     })
 }
 
+fn resolve_resource_config_for_rolegroup(
+    nifi: &NifiCluster,
+    rolegroup_ref: &RoleGroupRef<NifiCluster>,
+) -> Result<Resources<NifiStorageConfig, NoRuntimeLimits>> {
+    // Initialize the result with all default values as baseline
+    let conf_defaults = NifiConfig::default_resources();
+
+    // Retrieve global role resource config
+    let mut conf_role: Resources<NifiStorageConfig, NoRuntimeLimits> = nifi
+        .spec
+        .nodes
+        .as_ref()
+        .with_context(|| NoNodesDefinedSnafu {})?
```

Comment:
Seems like this should be perfectly legal (although not a particularly useful cluster).

Any reason not to `unwrap_or_default()` here?

Reply:
Primarily because there is no `impl Default for Role<T>`. And semantically I followed other places in the code that require `nodes`

Reply:
Ok, that should be fine for now then..